src/Microsoft.AspNetCore.Mvc.Razor/Compilation/CompiledViewManfiest.cs

```diff
@@ -15,19 +15,15 @@ public static class CompiledViewManfiest
 
         public static Type GetManifestType(AssemblyPart assemblyPart, string typeName)
         {
-            EnsureFeatureAssembly(assemblyPart);
-
-            var precompiledAssemblyName = new AssemblyName(assemblyPart.Assembly.FullName);
-            precompiledAssemblyName.Name = precompiledAssemblyName.Name + PrecompiledViewsAssemblySuffix;
-
-            return Type.GetType($"{typeName},{precompiledAssemblyName}");
```

Comment:
This returns null since switching to `Assembly.Load`. My guess is it's getting loaded in to a different load context. Regardless, since we already load the assembly by path, we can use that to look up types (and attributes in the future). Verified that this change fixes precompilation tests which are currently broken.